janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchIndex.java

```diff
@@ -926,6 +925,14 @@ public void restore(Map<String,Map<String, List<IndexEntry>>> documents, KeyInfo
                     return compat.boolMustNot(compat.match(fieldName, value));
                 } else if (predicate == Text.FUZZY || predicate == Text.CONTAINS_FUZZY) {
                     return compat.fuzzyMatch(fieldName, value);
+                } else if (predicate == Cmp.LESS_THAN) {
+                    return compat.lt(fieldName, value);
+                } else if (predicate == Cmp.LESS_THAN_EQUAL) {
+                    return compat.lte(fieldName, value);
+                } else if (predicate == Cmp.GREATER_THAN) {
+                    return compat.gt(fieldName, value);
+                } else if (predicate == Cmp.GREATER_THAN_EQUAL) {
+                    return compat.gte(fieldName, value);
                 } else
                     throw new IllegalArgumentException("Predicate is not supported for string value: " + predicate);
```

Comment:
Since you've added tests for all these cases (nice work!) would you also add a test to cover this case (e.g. test exception is thrown when predicate is not supported on String mapped string)? It's currently not covered in any of the indexing backends (e.g. [ElasticSearchIndex](https://codecov.io/gh/JanusGraph/janusgraph/src/8e2acfee4bbfd1c7caff92a0573af2248d89fded/janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchIndex.java#L937), [SolrIndex](https://codecov.io/gh/JanusGraph/janusgraph/src/8e2acfee4bbfd1c7caff92a0573af2248d89fded/janusgraph-solr/src/main/java/org/janusgraph/diskstorage/solr/SolrIndex.java#L722), etc.).

Reply:
I add a test for this case